### PR TITLE
Docs: how parameters become interface

### PR DESCRIPTION
**Pull request by hannahle**
@manske Let me know if there's anything part to the interface not listed here. 

@ayushkamat I see in bulk RNA-seq code there's `fork` inside the docstring for mutually exclusive parameters. Do we have anything like that for `LatchMetadata`?

### CODE REVIEW

File: docs/source/basics/customizing_interface.md [hunk 1]
```diff
@@ -111,9 +68,13 @@ The information given here will be rendered in the sidebar of the workflow in th
 * `repository`: A link to the Github repository where the code for the workflow lives,
 * `license`: The license that the workflow code falls under - must be a [SPDX](https://spdx.dev/) identifier.
 
+![Sidebar](../assets/ui/sidebar.png)
+
 ### Customizing Parameter Presentation
 
-When a workflow is registered, each parameter will receive a frontend component to ingest values in the browser. These components will perform HTML-native type validation on inputted values and can be customized from the python code. You can add a parameter to the interface by adding a `LatchParameter` object to your `LatchMetadata` object's parameter dictionary as below:
+When a workflow is registered, each workflow parameter will receive a frontend component to ingest values in the browser. 
+
```
Comment (manske) [inline docs/source/basics/customizing_interface.md:76]: I would go into more detail here about how parameters are added and specify that it has to match. Start it off with something like:

> Any input of the main @workflow function can be added & customized on the front end display for ingesting user values in the browser. To add a workflow parameter to the front end simply add a `LatchParameter` object to your `LatchMetadata` object's parameter dictionary:

Comment (manske) [reply docs/source/basics/customizing_interface.md:76]: And then add this part below the code snippet: 

> When a workflow is registered, each workflow parameter added to the `LatchMetadata` object will receive a frontend component to ingest values in the browser.

File: docs/source/basics/customizing_interface.md [hunk 2]
```diff
@@ -148,3 +109,292 @@ Each key in `metadata.parameters` must be the name of one of the parameters of t
 * `batch_table_column`: Whether this parameter should have a column to itself in the batch table at the top of the parameters page,
```
Comment (manske) [inline docs/source/basics/customizing_interface.md:109]: I would add the type each takes, something like: e.x. `batch_table_column`: (Boolean) Whether...

Comment (manske) [reply docs/source/basics/customizing_interface.md:109]: Also later we should add screenshots for most of how these correspond to the front end, i.e. display_name show the display name portion highlighted

File: docs/source/basics/customizing_interface.md [hunk 3]
```diff
@@ -148,3 +109,292 @@ Each key in `metadata.parameters` must be the name of one of the parameters of t
...
+from typing import Optional
+
+...
+@workflow
+def bactopia_wf(
+    ...
+    fastq_one: Optional[LatchFile] = None,
+    ...
+)
+```
```
Comment (manske) [inline docs/source/basics/customizing_interface.md:163]: I would separate the types and things like optional and default values — and explain them separately. But could do this later

File: docs/source/basics/customizing_interface.md [hunk 4]
```diff
@@ -148,3 +109,292 @@ Each key in `metadata.parameters` must be the name of one of the parameters of t
...
+
+Here, we are passing a list of `Sample`s as the input. On the Latch interface, when a user clicks the `+ Sample` button, a new block will be added with two parameters of the Python class `name` and `fastq`.
+
+---
+## Adding Documentation to your Workflow
+While most of the metadata of a workflow will be encapsulated in a LatchMetadata object, we still require a docstring in the body of the workflow function which specifies both a short and long-form description.
+
+
+
+### One Line Description
```
Comment (manske) [inline docs/source/basics/customizing_interface.md:285]: Shouldn't this description stuff all just be contained at the top in one place vs being repeated?

File: docs/source/basics/customizing_interface.md [hunk 5]
```diff
@@ -148,3 +109,292 @@ Each key in `metadata.parameters` must be the name of one of the parameters of t
...
+    """
+```
+
+![Long Description](../assets/ui/long-form%20description.png)
+
+---
+## Adding your workflow to a biological domain on Latch
+
+For public workflows, you may want to classify your workflow to a biological domains to make it easier for future users to discover. 
+
```
Comment (manske) [inline docs/source/basics/customizing_interface.md:389]: Think we should add list of common tags already used on the platform. And tell people to make sure they match their tags to tags that already exist on the platform if possible.

File: docs/source/basics/customizing_interface.md [hunk 6]
```diff
@@ -28,61 +28,18 @@ Then, use `latch preview` with the name of your workflow function:
...
 ```
 
-## Using `LatchMetdata` objects
-
-While most of the metadata of a workflow will be encapsulated in a `LatchMetadata` object, we still require a docstring in the body of the workflow function which specifies both a short and long-form description.
-
-### One Line Description
-
-The first line of the workflow function docstring will get rendered in the sidebar of the workflow and the workflow explore tab as a brief description of your workflow's functionality. Think of this as summarizing the entirety of your workflow's significance into a single line.
-
```
[the remark this answers is not in the document]
Comment (hannahle) [reply docs/source/basics/customizing_interface.md:38]: 1 sentence sounds good

**Pull request merged**